src/tox/package/builder/isolated.py

```diff
@@ -47,6 +50,17 @@ def build(config, session):
     return perform_isolated_build(build_info, package_venv, config.distdir, config.setupdir)
 
 
+def _ensure_importable_in_path(python_path, importable):
+    """Figure out if the importable exists in the given path."""
+    mod_chunks = importable.split(".")
+    pkg_path = python_path.join(*mod_chunks[:-1])
+    if not pkg_path.exists():
+        return False
+    if pkg_path.join(mod_chunks[-1], "__init__.py").exists():
```

Comment:
You should check here for pyc too. If you delete ``__init__.py`` but keep ``__init__.pyc`` the import will still work. Same below.

Reply:
I'm not sure if this is reasonable: the build backend is supposed to be committed along with the source and so it'd be a bad idea to encourage checking-in pre-compiled files... WDYT?

Reply:
Checking in the compiled files might be used/useful for obfuscation of the build-backend and to save space, not?

Reply:
Just a thought, but I consider this point nice to have. The rest though are non-negotiable.

Reply:
Well, I get the same impression as if I were to check-in a binary executable honestly... Though, it's not explicitly forbidden in PEP.

Reply:
Let's check what `pep517` does and follow the lead. Shall we?

Reply:
Sure.

Reply:
So `pypa/pep517` does not check the importable on the preparation stage (https://github.com/pypa/pep517/blob/3f1949a/pep517/wrappers.py#L131-L135) but it does so later after actually importing the backend (https://github.com/pypa/pep517/blob/3f1949a/pep517/_in_process.py#L90-L95) by checking `__file__` against those paths.

Also, I'd like to point out that PEP's backend-path section implies that it facilitates the src code use-case explicitly.

> In certain circumstances, projects may wish to include *the source code* for the build backend directly in the source tree

Given this, I'm not sure what to do. Any preferences? I should probably drop this part of the check here.

Reply:
@gaborbernat I've unresolved this thread (you managed to close it while I went out to search for clues in `pep517` and in the PEP text). I'd like to agree on what's better and I'm not completely against your idea, just trying to be as precise as possible wrt to what PEP says.

Reply:
I'm happy to do just as much as pep517 does here.